CORE-48075 - fetch namespaces from platform

ninaceban opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

Now for the Identity Map data element, the user should enter the namespace code manually.
We should fetch namespaces from the platform and expose them in a dropdown. If there are no namespaces or the call failed we should allow the user to enter namespace code manually.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All tests pass and I've made any necessary test changes.
- [ ] I've updated the schema in extension.json or no changes are necessary.

garyr commented on src/view/dataElements/identityMap/fetchNamespaces.js:
```diff
@@ -0,0 +1,30 @@
+export default ({ orgId, imsAccess }) => {
+  const headers = getBaseRequestHeaders({ orgId, imsAccess });
+
+  return fetch(`${platform.getHost()}/data/core/idnamespace/identities`, {
```
Hey @ninaceban, we will want to include the change that I just made in #108 to this

ninaceban replied:
thank you! I will update with latest

jfkhoury commented on src/view/dataElements/identityMap/fetchNamespaces.js:
```diff
@@ -0,0 +1,30 @@
+
+      return response.json();
+    })
+    .then(responseBody => responseBody);
```
Should we add a `catch` and return an empty `[]` as well?

jonsnyder commented on src/view/dataElements/identityMap.jsx:
```diff
@@ -43,9 +44,29 @@ const getInitialValues = ({ initInfo }) => {
+    imsAccess: initInfo.tokens.imsAccess
+  }).then(response => {
+    if (response.length > 0) {
+      const namespaces = response.map(namespace => {
```
It's cleaner to remove the return statement:
const namespaces = response.map(namespace => ({
  value: namespace.code,
  label: namespace.code
}));

ninaceban replied:
yes, thank you

ninaceban commented:
@jonsnyder and @Aaronius thank you for your suggestions, I have added more changes to the PR, could you please re-review it.
As per platform namespaces documentation, a namespace cannot be removed and a namespace code will never change once it was created.

jonsnyder reviewed (changes requested):
Here are a few UI suggestions after having run the sandbox for this branch:
* On initial load, expand the first accordion
* After a new identity is added, expand the accordion of the newly added identity
* Sort the namespaces in the drop down alphabetically
* Filter out ECID as a namespace from the drop down list
* Should we add some text to explain that ECID will be added automatically?
* When loading settings, sort the identities alphabetically by namespace
* Add an Infotip on the Authenticated State drop down
* Handle multiple identities with the same namespace. Either throw a validation error or generate the settings object correctly (currently, the second identity overwrites the first)
* Does there need to be any validation on the Primary checkbox? (I'm thinking that only one identifier should be primary)

jfkhoury commented:
@ninaceban Is this PR ready to be merged?

ninaceban commented:
@jonsnyder I have addressed most of your comments/suggestions except for those related to validation since `one primary ` validation if part of another PR.
Could you please take another look at the PR? Thank you

jonsnyder reviewed (approved):
Thank you. Looks great.
